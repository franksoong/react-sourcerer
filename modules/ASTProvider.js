import React from 'react';
import {parse} from 'babylon';

class ASTProvider extends React.Component {
  getChildContext() {
    return {
      ast: parse(this.props.source),
    };
  }

  render() {
    return (
      React.Children.only(this.props.children)
    );
  }
}

ASTProvider.childContextTypes = {
  ast: React.PropTypes.object,
};

export default ASTProvider;
