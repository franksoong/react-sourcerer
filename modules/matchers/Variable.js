import React from 'react';
import Match from '../Match';
import * as t from 'babel-types';

class Variable extends React.Component {
  render() {
    return (
      <Match
        {...this.props}
        match={(path) => {
          const {name} = this.props;
          const isDeclarator = t.isVariableDeclarator(path);
          if (!isDeclarator) return false;
          if (name && name !== path.node.id.name) return false;
          return isDeclarator;
        }}
      />
    );
  }
}

export default Variable;
