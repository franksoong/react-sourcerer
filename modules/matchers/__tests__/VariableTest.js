import ASTProvider from '../../ASTProvider';
import React from 'react';
import Variable from '../Variable';

import renderer from 'react-test-renderer';

describe('Variable', () => {
  it('should match a variable', () => {
    const output = renderer.create(<ASTProvider source={`
      const name = 'Merrick';
    `}>
      <Variable>
        {
          ((matches) => {
            return (
              <div>
                {matches.map((path) => path.node.type)}
              </div>
            );
          })
        }
      </Variable>
    </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });

  it('should filter a variable by name', () => {
    const output = renderer.create(<ASTProvider source={`
      const name = 'Merrick';
      const age = 26;
    `}>
      <Variable name='age'>
        {
          ((matches) => {
            return (
              <div>
                {matches.map((path) => path.node.id.name)}
              </div>
            );
          })
        }
      </Variable>
    </ASTProvider>);

    expect(output.toJSON()).toMatchSnapshot(); 
  });
});
