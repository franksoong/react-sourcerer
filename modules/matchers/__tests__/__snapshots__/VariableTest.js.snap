exports[`Variable should filter a variable by name 1`] = `
<div>
  age
</div>
`;

exports[`Variable should match a variable 1`] = `
<div>
  VariableDeclarator
</div>
`;
